backend/app/services/context/context_service.py

```
@@ -0,0 +1,695 @@
+            StorageError: If storage operation fails
+        """
+        # Get file extension
+        _, extension = os.path.splitext(filename)
+        extension = extension.lower()
+
+        # Validate file extension
+        if not self.parser.is_supported_extension(extension):
+            raise ValueError(
+                f"Unsupported file type: {extension}. "
+                f"Supported types: {', '.join(self.parser.SUPPORTED_EXTENSIONS.keys())}"
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)